crates/proto/src/rr/domain/label.rs

```diff
@@ -63,9 +63,14 @@ impl Label {
         match idna::Config::default()
             .use_std3_ascii_rules(true)
             .transitional_processing(true)
-            .verify_dns_length(true)
+            // checked manually after it's okay domain is already processed.
+            // and Errors is... opaque.
+            .verify_dns_length(false)
             .to_ascii(s)
         {
+            Ok(puny) if puny.len() > 63 => {
```

Comment:
Given that we're passing the punycode to `from_ascii()` anywhere, I think this match arm is pointless.

Reply:
That was my question, if already in punycode an early fail with `LabelBytesTooLong` error is better like in ascii. I can remove this and check if all call expect punycode.